### DISCUSSION TITLE
Custom Idle Timer 11.0.0-rc1.1

### DISCUSSION THREAD

**Discussion by Tulesha** [AvaloniaUI/Avalonia#11607] [Q&A] [answered]
Hello.
Since release 11.0.0-rc1.1, InputManager in Application has been made internal: [pull request](https://github.com/AvaloniaUI/Avalonia/pull/11557)

We have an IdleTimer in our application that monitors the idle state of the application. We used it to check user activity. Simillar to discord user status.
Here is the code from 11.0.0-preview8:
``` C#
public IdleTimer()
{
    m_idleInterval = TimeSpan.FromMilliseconds( IDLE_TIMER_INTERVAL );
    m_inputInterval = TimeSpan.FromMilliseconds( INPUT_TIMER_INTERVAL );

    Application.Current.InputManager.PostProcess.Subscribe( OnInput );

    StartTimer();
}
```

Can you please suggest how to implement this now?

**Comment by timunie** [comment]
to add: the idea behind this timer was to show user status, like Teams does it if you are awfk for longer time.

**Comment by kekekeks** [comment, marked as answer]
For monitoring the idle state you can use `InputElement.PointerMovedEvent.Raised.Subscribe(...)`
